plone/outputfilters/filters/resolveuid_and_caption.py

```diff
@@ -157,6 +157,44 @@ def __call__(self, data):
                     actual_url = relative_root.absolute_url()
                     href = urljoin(actual_url + '/', subpath) + appendix
                 attributes['href'] = href
+        for elem in soup.find_all(['source', 'iframe']):
+            # SOURCE is used for video and audio.
+            # IFRAME is used to embed PDFs.
+            attributes = elem.attrs
+            attrn = 'src'
+            src = attributes.get(attrn)
+            # an 'a' anchor element has no href
+            if not src:
+                attrn = 'srcset'
+                src = attributes.get(attrn)
+                if not src:
+                    continue
+            url_parts = urlsplit(src)
+            scheme = url_parts[0]
+            # we are only interested in path and beyond /foo/bar?x=2#abc
+            path_parts = urlunsplit(['', ''] + list(url_parts[2:]))
+            if not src.startswith('mailto<') \
```

Comment:
I am pretty sure the case from a href does not apply here.

Reply:
Reasonable!  Will fix.